modules/sdk/gradle-plugins/src/main/java/com/liferay/gradle/plugins/LiferayOSGiPlugin.java

```diff
@@ -504,8 +508,65 @@ public void execute(Task task) {
 		return directDeployTask;
 	}
 
+	private ExecuteBndTask _addTaskBuildWSDDExecuteBnd(
+		final Project project, BuildWSDDTask buildWSDDTask) {
+
+		ExecuteBndTask executeBndTask = GradleUtil.addTask(
+			project, buildWSDDTask.getName() + "ExecuteBnd",
+			ExecuteBndTask.class);
+
+		executeBndTask.dependsOn(buildWSDDTask);
+
+		executeBndTask.setBaseDir(
+			new Callable<File>() {
+
+				@Override
+				public File call() throws Exception {
+					return BundleUtils.getBase(project);
+				}
+
+			});
+
+		executeBndTask.setClasspath(
+			project.files(
+				new Callable<Object[]>() {
+
+					@Override
+					public Object[] call() throws Exception {
+						SourceSet sourceSet = GradleUtil.getSourceSet(
+							project, SourceSet.MAIN_SOURCE_SET_NAME);
+
+						SourceSetOutput sourceSetOutput = sourceSet.getOutput();
+
+						return new File[] {
+							sourceSetOutput.getClassesDir(),
+							sourceSetOutput.getResourcesDir()
+						};
+					}
+
+				}));
+
+		executeBndTask.setFailOnError(true);
+		executeBndTask.setResourceDirs(project.files());
+
+		executeBndTask.setSourceDirs(
+			project.files(
+				new Callable<Object[]>() {
+
+					@Override
+					public Object[] call() throws Exception {
+						return BundleUtils.getSources(project);
+					}
+
+				}));
+
+		return executeBndTask;
+	}
+
 	private Jar _addTaskBuildWSDDJar(
```

Comment:
Why are you keeping this? The idea is to replace `buildWSDDJar` of type `Jar` with the new `buildWSDDJar` of type `ExecuteBndTask`.